components/dotaReplay.py

```diff
@@ -2,52 +2,164 @@
 import shelve
 from datetime import datetime
 
-async def check_recent_matches(channel, player_list, OPENDOTA_API_KEY):
-    s = shelve.open('./database/george_token_count.db')
-    if s.get("token_count") is None:
-        s["token_count"] = 0
 
+m_open_dota_api_key = ""
+
+def get_arg_list(message, expected_num_args: int, strict: bool):
+    arg_list = message.content.split()
+    for item in arg_list:
+        print(item)
+    # ignore command
+    if strict:
+        arg_list = arg_list[1:expected_num_args+1]
+        print(arg_list)
+        if len(arg_list) != expected_num_args:
+            return []
+    else:
+        arg_list = arg_list[1:]
+        if len(arg_list) < expected_num_args:
+            return []
+
+    return arg_list
+
+
+async def send_message(channel, msg: str):
+    try:
+        await channel.send(msg)
+    except Exception as e:
+        print(e)
+
+
+def authorIsAdmin(author, admin_role: str):
+    is_admin = False
+    for role in author.roles:
+        temp_admin_role = admin_role
+        if role.name[0] != "@":
+            temp_admin_role = admin_role[1:]
+        if temp_admin_role == role.name:
+            is_admin = True
+    
+    return is_admin
+
+
+async def add_player(message, admin_role):
+    channel = message.channel
+
+    if not authorIsAdmin(message.author, admin_role):
+        await send_message(channel, "Sorry, you need to be a dictator to use this command.")
+        return
+
+    invalid_message = "Invalid arguments. One should be a string of characters (no spaces), and one should be a number."
+    # string parsing
+    arg_list = get_arg_list(message, 2, True)
+    print (arg_list)
+    if not arg_list:
+        await send_message(channel, invalid_message)
+        return
+
+    first_arg = arg_list[0]
+    second_arg = arg_list[1]
+    print(first_arg)
+    print(second_arg)
+
+    if not((first_arg.isdigit() and second_arg.isalpha()) or (first_arg.isalpha() and second_arg.isdigit())):
+        await send_message(channel, invalid_message)
+        return
+            
+    # shelf: {key = player id, value = player irl name}
+    player_id = first_arg if first_arg.isdigit() else second_arg
+    player_name = second_arg if first_arg.isdigit() else first_arg
+
+    player_list_shelf = shelve.open('./database/dota_player_list.db')
+    if player_list_shelf.get(player_id) is None:
+        # add
+        if player_id and player_name:
+            player_list_shelf[player_id] = player_name
```

Comment:
Should we not be validating player_id and ensure the account exists?

Reply:
by validating you mean a "real check" (i.e. requesting from opendota and checking if its valid) or a fake check (i.e. check # of digits in string)? This is only gonna be used by us so functionality wise we don't even really need a check, but if we want it to be more robust I can do either

Reply:
Was talking about real check. Not as important if we aren't opening up the command to everyone tho.